torch/fx/experimental/Partitioner.py

```diff
@@ -156,6 +271,80 @@ def find_single_partition(self) -> None:
         partition_0.add_parent(self.root_partition)
         return
 
+    def size_based_partition(self, available_mem: Union[float, int]) -> None:
+        """This method partitions the graph based on memory size.
+           We assume all devices have the same memory size.
+           The basic idea is:
+           First, create a new partition.
+           Then travel the graph in the order of bfs level.
+           The travel only focuses on op nodes
+           (call_function, call_module, call_method)
+           Check if a op node and its input nodes
+           could fit into the current partition.
+           If the current partition is full, create a new one
+           and continue traveling nodes.
+           For orphan nodes (placeholder or constants are never used),
```

Comment:
Can you explain the behavior when placeholders or constants are encountered in the graph?